Persistence/Repositories/FinancialYearRepository.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Domain;
+
+using Microsoft.EntityFrameworkCore;
+
+namespace Persistence.Repositories;
+
+public class FinancialYearRepository : IFinancialYearRepository
+{
+    
+    private readonly HaSpManContext _context;
+
+    public FinancialYearRepository(IDbContextFactory<HaSpManContext> contextFactory)
+    {
+
+        _context = contextFactory.CreateDbContext();
+    }
+
+    public Task<FinancialYear?> GetById(Guid id, CancellationToken cancellationToken)
+    {
+        return _context.FinancialYears.FirstOrDefaultAsync(x => x.Id == id, cancellationToken);
+    }
+
+    public void Add(FinancialYear financialYear)
+    {
+        _context.FinancialYears.Add(financialYear);
+    }
+
+    public async Task SaveChangesAsync(CancellationToken cancellationToken)
+    {
+        await _context.SaveChangesAsync(cancellationToken);
+    }
+}
+
+public interface IFinancialYearRepository
```

Comment:
This interface should be in the Domain project. See https://stackoverflow.com/a/72399927 for reasons why